llvm/test/MC/Disassembler/AMDGPU/gfx12_dasm_sop1.txt

```diff
@@ -1,5 +1,14 @@
 # RUN: llvm-mc -triple=amdgcn -mcpu=gfx1200 -disassemble -show-encoding < %s | FileCheck -check-prefixes=GFX12 %s
 
+# GFX12: s_alloc_vgpr 0x1235                          ; encoding: [0xff,0x53,0x80,0xbe,0x35,0x12,0x00,0x00]
+0xff,0x53,0x80,0xbe,0x35,0x12,0x00,0x00
+
+# GFX12: s_alloc_vgpr 18                              ; encoding: [0x92,0x53,0x80,0xbe]
+0x92,0x53,0x80,0xbe
+
+# GFX12: s_alloc_vgpr s35                             ; encoding: [0x23,0x53,0x80,0xbe]
+0x23,0x53,0x80,0xbe
+
```

Comment:
Does this need error tests for VGPRs, or the exotic scalar registers?

Reply:
Done. I'm not sure which ones are the exotic SGPRs, I added tests for m0, scc, vcc and exec. We don't accept the last 2 (although I think sp3 accepts all of them).